service.py

```diff
@@ -98,7 +99,12 @@ def get_study_kf_id(self, study_id):
             return
         if study_id in self.external_ids:
             return self.external_ids[study_id], self.version[study_id]
-        resp = requests.get(self.api+'/studies?external_id='+study_id)
+        while True:
```

Comment:
What happens if the lambda times out while this is retrying?

Reply:
studies are already filtered out by invoker for issues like version numbers, non dbgap studies etc. also we are giving enough time to update all fields otherwise it triggers new lambda.

Reply:
> studies are already filtered out by invoker for issues like version numbers, non dbgap studies etc.

not relevant here

> also we are giving enough time to update all fields

That's not clear to me.
The outer loop will let you begin a new sequence if there are 50 seconds left, and then you have many new inner loops with no time checks in them. What do you do if something else is overloading the server and the inner loops don't complete?

> otherwise it triggers new lambda.

That doesn't happen here. That only happens outside of updater.update_acl in the def handler event records loop. But now you have loops inside of updater.update_acl. What happens if you start updater.update_acl and then loop and loop inside and any of the inside loops don't finish?